### DISCUSSION TITLE
Add Quick Sync support

### DISCUSSION THREAD

**Discussion by Mishasama** [Xaymar/obs-stream-effects#849] [Ideas & Feedback]
### Explain the Feature/Enhancement

The QS encoder in the OBS looks lacks the functions like NVENC.
I hope we can do much more tweaks for QS.

### What makes this feature/enhancement useful for StreamFX and its users?

The laptop version iGPU is very powerful. And the Intel graphics card is coming.
So we should have a way to release the QS all the power with FFmpeg.

### Checklist

- [ ] I am offering a monetary bounty/reward for the implementation of this Feature/Enhancement.
- [X] There is no other way to currently do this in StreamFX or OBS Studio.

### Any additional Information we need to know?

It's better to add support for both H264 and H265.

**Comment by Xaymar** [comment]
QSV in OBS is maintained by Intel. There is nothing that can be offered that isn't already offered, as QSV does not have that much in terms of configuration.

**Comment by Mishasama** [comment]
That is still something that can be configured by [FFmpeg](https://gist.github.com/Mishasama/7e99a08be0f511881c2398897cf682b5#file-h264_qsv-txt) that was hidden by OBS.
e.g. Look ahead

We may mix them to find the best optimization for our scenarios.

**Comment by Xaymar** [comment]
Many of the options FFmpeg offers are from the previous API version, which have no effect on 11th generation and up, as well as the new architecture. But sure, it could be offered - even if the integration in OBS Studio already allows for the highest possible quality QSV can offer.

**Comment by Mishasama** [comment]
> Many of the options FFmpeg offers are from the previous API version, which have no effect on 11th generation and up, as well as the new architecture. But sure, it could be offered - even if the integration in OBS Studio already allows for the highest possible quality QSV can offer.

The problem for me is, that I don't know what value it uses for its specific parameters. And I can't specifically optimize parameters for different types of content.
e.g., FPS games & RTS games
I can't just use the highest setting because it will overload.
So, I have to find out the best value for each scene — maximum quality without overload.

Do you have any suggestions for this situation?

**Comment by Xaymar** [comment]
Tracking this as: https://github.com/Xaymar/obs-StreamFX/discussions/858
